### PR TITLE
Fix documents for development and bug report

### PR DESCRIPTION
**Pull request by wxiaoguang**
This PR fixes some documents and sentences.

1. Tell users to `Make sure you are using the latest release` before submitting an issue.
2. Update `hacking-on-gitea.en-us.md` to tell developers to do lint correctly.
3. Remove out-dated information, eg `make fmt-check`
4. Fix some typos.

Hopefully we can keep documents correct and up-dated.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yaml [hunk 1]
```diff
@@ -11,15 +11,16 @@ body:
       1. Please speak English, this is the language all maintainers can speak and write.
       2. Please ask questions or configuration/deploy problems on our Discord
          server (https://discord.gg/gitea) or forum (https://discourse.gitea.io).
-      3. Please take a moment to check that your issue doesn't already exist.
+      3. Make sure you are using the latest release and
+         take a moment to check that your issue hasn't been reported before.
```
Comment (delvh) [inline .github/ISSUE_TEMPLATE/bug-report.yaml:15]: I'd say these two can be separate points:
```suggestion
      3. Please make sure you are using the latest release
      4. Please take a moment to check that your issue hasn't been reported before.
```

File: .github/ISSUE_TEMPLATE/bug-report.yaml [hunk 2]
```diff
@@ -11,15 +11,16 @@ body:
...
       4. Make sure it's not mentioned in the FAQ (https://docs.gitea.io/en-us/faq)
```
Comment (delvh) [inline .github/ISSUE_TEMPLATE/bug-report.yaml:16]:
```suggestion
      4. Please make sure it's not mentioned in the FAQ (https://docs.gitea.io/en-us/faq)
```

**Review by lunny — APPROVED** [review]

File: docs/content/doc/developers/hacking-on-gitea.en-us.md [hunk 3]
```diff
@@ -29,17 +29,16 @@ required to build the JavaScript and CSS files. The minimum supported Node.js
...
 on the executable path. If you don't add the go bin directory to the
 executable path you will have to manage this yourself.
 
-**Note 2**: Go version {{< min-go-version >}} or higher is required; however, it is important
-to note that our continuous integration will check that the formatting of the
-source code is not changed by `gofmt` using `make fmt-check`. Unfortunately,
-the results of `gofmt` can differ by the version of `go`. It is therefore
+**Note 2**: Go version {{< min-go-version >}} or higher is required.
+Gitea uses `gofmt` to format source code. However, the results of 
+`gofmt` can differ by the version of `go`. Therefor it is
```
Comment (delvh) [inline docs/content/doc/developers/hacking-on-gitea.en-us.md]:
```suggestion
`gofmt` can differ by the version of `go`. Therefore, it is
```

File: docs/content/doc/developers/hacking-on-gitea.en-us.md [hunk 4]
```diff
@@ -149,25 +148,26 @@ On macOS, watching all backend source files may hit the default open files limit
...
-make fmt-check # which runs make fmt internally
+# lint both frontend and backend code
+make lint
+# lint only backend code
+make lint-backend
 ```
 
-**Note**: The results of `go fmt` are dependent on the version of `go` present.
+**Note**: The results of `gofmt` are dependent on the version of `go` present.
 You should run the same version of go that is on the continuous integration
```
Comment (delvh) [inline docs/content/doc/developers/hacking-on-gitea.en-us.md:169]:
```suggestion
You should run the same version of go that is used by the CI.
```

File: docs/content/doc/developers/hacking-on-gitea.en-us.md [hunk 5]
```diff
@@ -149,25 +148,26 @@ On macOS, watching all backend source files may hit the default open files limit
...
-server as mentioned above. `make fmt-check` will only check if your `go` would
-format differently - this may be different from the CI server version.
+server as mentioned above.
```
Comment (delvh) [inline docs/content/doc/developers/hacking-on-gitea.en-us.md:170]:
```suggestion
```

File: docs/content/doc/developers/hacking-on-gitea.en-us.md [hunk 6]
```diff
@@ -149,25 +148,26 @@ On macOS, watching all backend source files may hit the default open files limit
 
 ### Formatting, code analysis and spell check
 
-Our continuous integration will reject PRs that are not properly formatted, fail
-code analysis or spell check.
+Our continuous integration will reject PRs that fail the code linters (including format check, code analysis and spell check).
 
-You should format your code with `go fmt` using:
+You should format your code:
```
Comment (delvh) [inline docs/content/doc/developers/hacking-on-gitea.en-us.md:153]:
```suggestion
You should format your code using
```

File: docs/content/doc/developers/hacking-on-gitea.en-us.md [hunk 7]
```diff
@@ -149,25 +148,26 @@ On macOS, watching all backend source files may hit the default open files limit
...
 
 ```bash
 make fmt
 ```
 
-and can test whether your changes would match the results with:
+and lint the source code:
```
Comment (delvh) [inline docs/content/doc/developers/hacking-on-gitea.en-us.md:159]:
```suggestion
and lint it using
```

File: .github/ISSUE_TEMPLATE/bug-report.yaml (hunk 1 above)
Comment (wxiaoguang) [reply .github/ISSUE_TEMPLATE/bug-report.yaml:15]: There are already many `please`s above and below. I don't like to put too many pleases. `Make sure` is clear enough

File: docs/content/doc/developers/hacking-on-gitea.en-us.md (hunk 5 above)
Comment (wxiaoguang) [reply docs/content/doc/developers/hacking-on-gitea.en-us.md:170]: The original is correct.

File: docs/content/doc/developers/hacking-on-gitea.en-us.md (hunk 4 above)
Comment (wxiaoguang) [reply docs/content/doc/developers/hacking-on-gitea.en-us.md:169]: The original is not wrong.

File: docs/content/doc/developers/hacking-on-gitea.en-us.md (hunk 6 above)
Comment (wxiaoguang) [reply docs/content/doc/developers/hacking-on-gitea.en-us.md:153]: The original text is better.

File: docs/content/doc/developers/hacking-on-gitea.en-us.md (hunk 7 above)
Comment (wxiaoguang) [reply docs/content/doc/developers/hacking-on-gitea.en-us.md:159]: The original text is better.

File: docs/content/doc/developers/hacking-on-gitea.en-us.md (hunk 6 above)
Comment (delvh) [reply docs/content/doc/developers/hacking-on-gitea.en-us.md:153]: Okay

**Review by singuliere — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**